packages/linear-progress/README.md

```diff
@@ -1,64 +1,134 @@
+> NOTE: The Material Web Components are distributed as ES2017 JavaScript
+> Modules, and use the Custom Elements API. They are compatible with all modern
+> browsers including Chrome, Firefox, Safari, Edge, and IE11, but an additional
+> tooling step is required to resolve *bare module specifiers*, as well as
+> transpilation and polyfills for Edge and IE11. See
+> [here](https://github.com/material-components/material-components-web-components#quick-start)
+> for detailed instructions.
 
-* When you're ready to use mwc-linear-progress in a project, install it via [npm](https://www.npmjs.com/). To run the project in the browser, a module-compatible toolctain is required. We recommend installing the [Polymer CLI](https://github.com/Polymer/polymer-cli) and using its development server as follows.
+## Example usage
 
-  1. Ensure the webcomponents polyfills are included in your HTML page
+### Determinate
 
-      - Install webcomponents polyfills
+<img src="images/determinate.png" height="34px">
 
-          ```npm i @webcomponents/webcomponentsjs```
+```html
+<script type="module">
+  import '@material/mwc-linear-progress';
+</script>
+<mwc-linear-progress progress="0.5"></mwc-linear-progress>
+```
 
-      - Add webcomponents polyfills to your HTML page
+### Indederminate
 
-          ```<script src="@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>```
+<img src="images/indeterminate.gif" height="77px">
 
-  1. Add mwc-linear-progress to your project:
+```html
+<mwc-linear-progress indeterminate></mwc-linear-progress>
+```
 
-      ```npm i @material/mwc-linear-progress```
+### Determinate buffer
 
-  1. Import the mwc-linear-progress definition into your HTML page:
+<img src="images/determinate-buffer.gif" height="45px">
 
-      ```<script type="module" src="@material/mwc-linear-progress/index.js"></script>```
+```html
+<mwc-linear-progress progress="0.25" buffer="0.5"></mwc-linear-progress>
+```
 
-      Or into your module script:
+### Reversed
 
-      ```import {LinearProgress} from "@material/mwc-linear-progress"```
+<img src="images/reversed.gif" height="45px">
 
-  1. Create an instance of mwc-linear-progress in your HTML page, or via any framework that [supports rendering Custom Elements](https://custom-elements-everywhere.com/):
 
-      ```<mwc-linear-progress determinate progress="0.5" buffer="1"></mwc-linear-progress>```
+```html
+<mwc-linear-progress
+    reverse
+    progress="0.25"
+    buffer="0.5">
+</mwc-linear-progress>
+```
 
-  1. Install the Polymer CLI:
+### Styled
 
-      ```npm i -g polymer-cli```
+<img src="images/styled.gif" height="45px">
 
-  1. Run the development server and open a browser pointing to its URL:
+```html
+<style>
+  mwc-linear-progress {
+    --mdc-theme-primary: red;
+    --mdc-linear-progress-buffer-color: orange;
+    /* Note: all that was changed from default was "fill='orange'" */
+    --mdc-linear-progress-buffering-dots-image:
```

Comment:
Not needed for this PR but is this really the only way to change the color of the dots? Could we use a CSS Custom Property in the SVG instead? Does ShadyCSS support that?

(Also let's check if the new linear progress design still has the dots before doing anything.)

Reply:
It's a background-image so there is no dom we can edit >.<

What they do in the SASS is that they use SASS to convert the color to url format, and then they just inject it into the url string. The problem is that, with CSS CPs, we cannot do:

```
background-image: url('.... color=var(--url-encoded-color, '%23FF00FF') ...');
```

Reply:
alternatively, they can have actual SVG files with different colors and just change the `--mdc-linear-progress-buffering-dots-image` to point to the new file